packages/firestore/src/api/database.ts

```diff
@@ -1566,10 +1574,19 @@ export class Query implements firestore.Query {
     };
   }
 
-  get(): Promise<firestore.QuerySnapshot> {
-    validateExactNumberOfArgs('Query.get', arguments, 0);
+  get(options?: firestore.GetOptions): Promise<firestore.QuerySnapshot> {
+    validateBetweenNumberOfArgs('Query.get', arguments, 0, 1);
     return new Promise(
       (resolve: Resolver<firestore.QuerySnapshot>, reject: Rejecter) => {
+        if (options && options.source === 'cache') {
+          this.firestore
+            .ensureClientConfigured()
+            .getDocumentsFromLocalCache(this, {
+              next: (snap: firestore.QuerySnapshot) => resolve(snap),
+              error: reject
```

Comment:
I think we want to resolve with an empty snapshot here?

Reply:
Actually, it looks like as-implemented, error can't be called so this is fine. I think this will be clearer if we implement it with a Promise return rather than Observer in any case...